internal/states/remote/remote_grpc.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package remote
+
+import (
+	"github.com/hashicorp/terraform/internal/providers"
+	"github.com/hashicorp/terraform/internal/states/statemgr"
+)
+
+// NewRemoteGRPC returns a remote state manager (remote.State) containing
+// an implementation of remote.Client that allows Terraform to interact with
+// a provider implementing pluggable state storage.
+//
+// The remote.Client implementation's methods invoke the provider's RPC
+// methods to perform tasks like reading in state, locking, etc.
+//
+// NewRemoteGRPC requires these arguments to create the remote.Client:
+// 1) the provider interface, needed to call gRPC methods
+// 2) the name of the state storage implementation in the provider
+// 3) the name of the state/the active workspace
+func NewRemoteGRPC(provider providers.Interface, typeName string, stateId string) statemgr.Full {
+	mgr := &State{
+		Client: &grpcClient{
+			provider: provider,
+			typeName: typeName,
+			stateId:  stateId,
+		},
+	}
+	return mgr
+}
+
+var (
+	_ Client       = &grpcClient{}
+	_ ClientLocker = &grpcClient{}
+	// TODO(SarahFrench/radeksimko - do we need to implement the interface below?)
+	// _ ClientForcePusher = &grpcClient{}
```

Comment:
As far as I can tell this is only used in the context of TFC currently and not in any of the existing state stores.

So if that is the case I would leave it out.